apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+
+        /// <summary>
+        /// Compares two values for equality.
+        /// </summary>
+        /// <param name="lhs">The first value to compare.</param>
+        /// <param name="rhs">The second value to compare.</param>
+        /// <returns><c>true</c> if the two arguments are equal; <c>false</c> otherwise.</returns>
+        public static bool operator ==(SpannerDate lhs, SpannerDate rhs) => lhs.Equals(rhs);
+
+        /// <summary>
+        /// Compares two values for inequality.
+        /// </summary>
+        /// <param name="lhs">The first value to compare.</param>
+        /// <param name="rhs">The second value to compare.</param>
+        /// <returns><c>false</c> if the two arguments are equal; <c>true</c> otherwise.</returns>
+        public static bool operator !=(SpannerDate lhs, SpannerDate rhs) => !lhs.Equals(rhs);
+
+        /// <summary>Compares two values to determine which is less.</summary>
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
+        /// <param name="rhs">The value to compare with <paramref name="rhs" />.</param>
+        /// <returns><c>true</c> if <paramref name="lhs" /> is less than <paramref name="rhs" />; otherwise, <c>false</c>.</returns>
+        public static bool operator <(SpannerDate lhs, SpannerDate rhs) => lhs.CompareTo(rhs) < 0;
+
+        /// <summary>Compares two values to determine which is less or equal.</summary>
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
+        /// <param name="rhs">The value to compare with <paramref name="rhs" />.</param>
+        /// <returns><c>true</c> if <paramref name="lhs" /> is less than or equal to <paramref name="rhs" />; otherwise, <c>false</c>.</returns>
+        public static bool operator <=(SpannerDate lhs, SpannerDate rhs) => lhs.CompareTo(rhs) <= 0;
+
+        /// <summary>Compares two values to determine which is greater.</summary>
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
+        /// <param name="rhs">The value to compare with <paramref name="rhs" />.</param>
+        /// <returns><c>true</c> if <paramref name="lhs" /> is greater than <paramref name="rhs" />; otherwise, <c>false</c>.</returns>
+        public static bool operator >(SpannerDate lhs, SpannerDate rhs) => lhs.CompareTo(rhs) > 0;
+
+        /// <summary>Compares two values to determine which is greater or equal.</summary>
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
+        /// <param name="rhs">The value to compare with <paramref name="rhs" />.</param>
+        /// <returns><c>true</c> if <paramref name="lhs" /> is greater than or equal to <paramref name="rhs" />; otherwise, <c>false</c>.</returns>
+        public static bool operator >=(SpannerDate lhs, SpannerDate rhs) => lhs.CompareTo(rhs) >= 0;
+
+        /// <summary>Adds days to the SpannerDate to compute the new SpannerDate.</summary>
```

Comment:
I'd remove these three operators. I don't think adding an int to a date is a clear operation.

We can always have a PlusDays or AddDays method instead. (Personally I prefer PlusDays for clarity, but AddDays would be consistent with DateTime)